docs/create-packages/set-package-type.md

```diff
@@ -9,28 +9,52 @@ ms.topic: conceptual
 
 # Set a NuGet package type
 
-With NuGet 3.5+, packages can be marked with a specific *package type* to indicate its intended use. Packages not marked with a type, including all packages created with earlier versions of NuGet, default to the `Dependency` type.
+Packages can be marked with one more more *package types* to indicate its intended use. Packages not marked with a type, including all packages created with earlier versions of NuGet, default to the `Dependency` type.
 
 - `Dependency` type packages add build- or run-time assets to libraries and applications, and can be installed in any project type (assuming they are compatible).
 
-- `DotnetTool` type packages are extensions to the [dotnet CLI](/dotnet/articles/core/tools/index) and are invoked from the command line. Such packages can be installed only in .NET Core projects and have no effect on restore operations. More details about these per-project extensions are available in the  [.NET Core extensibility](/dotnet/articles/core/tools/extensibility#per-project-based-extensibility) documentation.
+- `DotnetTool` type packages are .NET tools that can be installed by the [dotnet CLI](/dotnet/articles/core/tools/index).
 
 - `Template` type packages provide [custom templates](/dotnet/core/tools/custom-templates) that can be used to create files or projects like an app, service, tool, or class library.
 
 - Custom type packages use an arbitrary type identifier that conforms to the same format rules as package IDs. Any type other than `Dependency` and `DotnetTool`, however, are not recognized by the NuGet Package Manager in Visual Studio.
 
-Package types are set in the `.nuspec` file. It's best for backwards compatibility to *not* explicitly set the `Dependency` type and to instead rely on NuGet assuming this type when no type is specified.
-
-- `.nuspec`: Indicate the package type within a `packageTypes\packageType` node under the `<metadata>` element:
-
-    ```xml
-    <?xml version="1.0" encoding="utf-8"?>
-    <package xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
-        <metadata>
-        <!-- ... -->
-        <packageTypes>
-            <packageType name="DotnetTool" />
-        </packageTypes>
-        </metadata>
-    </package>
-    ```
+> [!NOTE]
+> Support for package types was added in NuGet 3.5.
+> If you don't need a custom package type, it's best to *not* explicitly set the package type.
+> NuGet defaults to the `Dependency` type when no type is specified.
+
+# [Using dotnet CLI](#tab/dotnet)
+
+Package types can be set in the project file (`.csproj`):
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net5.0</TargetFramework>
+    
+    <PackageType>MyCustomType, 1.0.0.0;Dependency</PackageType>
```

Comment:
Does `MyCustomType, 1.0.0.0` get treated as a "single" string, or does it get treated as a versioned `MyCustomType`? How/where do the versions come into play?

Reply:
Currently a package with multiple package types can't be installed in the Package Manager UI. 

I would really prefer that we don't offer up example with multiple package types like this. 
The `Dependency` type doesn't really add any value here either. 

MyCustomType would not be installable in the PM UI, https://github.com/NuGet/NuGet.Client/blob/1328b1db6ef99fd0f0c7cf9201e4c75a366a88dd/src/NuGet.Core/NuGet.PackageManagement/InstallationCompatibility.cs#L140. 

I think that we should only offer up example of 1 package type and be explicit that those package types won't be installable through the PM UI. 

Note that dotnet add package won't do this check...sooo...complicated :D

Reply:
> Does `MyCustomType, 1.0.0.0` get treated as a "single" string, or does it get treated as a versioned `MyCustomType`? How/where do the versions come into play?

@zivkan, it gets parsed as a versioned `MyCustomType`. Package type names follow the same character limitations as package IDs so the comma and space would not be valid. The nuspec would look like this:

```xml
<package>
  <metadata>
    ...
    <packageTypes>
      <packageType name="MyCustomType" version="1.0.0.0" />
      <packageType name="Dependency" />
    </packageTypes>
    ...
  </metadata>
</package>
```

Side note, when the `Dependency` package type is parsed, it will get a version of `0.0`.

Reply:
> I think that we should only offer up example of 1 package type and be explicit that those package types won't be installable through the PM UI.

Seems fine to me. 

In my mind, an ideal situation would be if all client tools were more clear in their package type enforcement, per https://github.com/NuGet/Home/issues/10468. That is all fail for non-Dependency packages but Dependency + other package type combinations work everywhere. But before that DCR lands I agree perhaps the docs should guide people towards working scenarios.

Alternatively, we could link to the bug saying that this doesn't work in all NuGet client scenarios? I don't feel strongly one way or another.

Reply:
Thanks for the feedback y'all! I'll mark this as resolved as I:

* Added more details about delimiters and the package type format
* Removed the example where there was a custom package type plus the `Dependency` type
* Linked to https://github.com/NuGet/Home/issues/10468

Feel free to re-open this thread if you have additional suggestions